### PR TITLE
signup, select role, signin  api's

### PR DESCRIPTION
**Pull request by kkavya007**

### CODE REVIEW

File: Backend/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const {User} = require('../models/userModel');
+const signUp = async (req,res)=>{
+    try {
+        const {
+            username,
+            email,
+            password,
+            confirmPassword
+        }=req.body;
+        if(!username || !email || !password || !confirmPassword){
```
Comment (TirumaleswaraReddy) [inline Backend/controllers/authController.js:13]: Add space trim for every property.

File: Backend/node_modules/.package-lock.json
Comment (TirumaleswaraReddy) [inline Backend/node_modules/.package-lock.json:1]: Add these files (lock.json) to .gitignore file. Do not push nodemodule files.

File: Backend/package-lock.json
Comment (TirumaleswaraReddy) [inline Backend/package-lock.json:1]: Add these files (lock.json) to .gitignore file.

**Review by Thanu317 — APPROVED** [review]

**Review by Ganesh1101 — APPROVED** [review]

**Review by Thanu317 — APPROVED** [review]

File: Backend/controllers/authController.js (hunk 1 above)
Comment (Thanu317) [reply Backend/controllers/authController.js:13]: ok

File: Backend/package-lock.json
Comment (Thanu317) [reply Backend/package-lock.json:1]: lock required for locking the packages for feature references

**Pull request merged**